R/get_ref.R

```diff
@@ -96,7 +96,14 @@ get_ref_min.remote_ref_cran <- function(remote_ref, op = "", op_ver = "") {
   min_ver <- Filter(function(x) x == min(pv), pv)
 
   new_ref <- sprintf("%s@%s", remote_ref$ref, names(min_ver)) # @TODO deparse, add ver, parse again
-  pkgdepends::parse_pkg_ref(new_ref)
+  tryCatch(
+    pkgdepends::parse_pkg_ref(new_ref),
+    error = function(err) {
+      cli::cli_alert_danger(
+        "Problem finding version for: `{remote_ref$package} ({op} {op_ver})`"
+      )
+      stop(err)
+    })
```

Comment:
I would say that `pkgdepends::parse_pkg_ref()` is quite low-risk call hence I don't expect it to fail. Can you give me an example where you have an error, i.e. when tryCatch will be useful?

Reply:
This happens when there's an invalid version for a give package, let's say `tern (>= 10.8.3)`

The error when this happens is show below, which I don't think has a good description.

ps.: I'm changing the message to be `"Possible problem finding release for: {remote_ref$package} ({op} {op_ver})"`. It seems like a better description.

```R
Error in "pkgdepends::parse_pkg_ref(new_ref)" :
! `ref` must be a string (character scalar), but it is an empty character vector.
```

![image](https://github.com/insightsengineering/verdepcheck/assets/211358/37718980-c13c-45bd-89ca-da6269d7c87f)